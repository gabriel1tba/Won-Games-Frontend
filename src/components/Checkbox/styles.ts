import styled, { css } from 'styled-components';
import { CheckBoxProps } from '.';

export const Wrapper = styled.main``;

export const Label = styled.label<Pick<CheckBoxProps, 'labelColor'>>`
  ${({ theme, labelColor }) => css`
    color: ${theme.colors[labelColor!]};
  `}
`;
