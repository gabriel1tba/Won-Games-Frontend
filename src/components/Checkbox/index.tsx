import * as S from './styles';

export type CheckBoxType = {
  label?: string;
  labelFor?: string;
};

const Checkbox = ({ label, labelFor = '' }: CheckBoxType) => (
  <S.Wrapper>
    <input type="checkbox" id={labelFor} />
    {!!label && <label htmlFor={labelFor}>{label}</label>}
  </S.Wrapper>
);

export default Checkbox;
