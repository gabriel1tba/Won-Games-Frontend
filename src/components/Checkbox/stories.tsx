import { Story, Meta } from '@storybook/react/types-6-0';
import Checkbox, { CheckBoxProps } from '.';

export default {
  title: 'Checkbox',
  component: Checkbox,
} as Meta;

export const Default: Story<CheckBoxProps> = (args) => <Checkbox {...args} />;
