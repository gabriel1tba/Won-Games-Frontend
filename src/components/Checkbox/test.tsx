import { screen } from '@testing-library/react';
import renderWithTheme from 'utils/tests/helpers';

import Checkbox from '.';

describe('<Checkbox />', () => {
  it('should render the heading', () => {
    const { container } = renderWithTheme(<Checkbox />);

    expect(
      screen.getByRole('heading', { name: /Checkbox/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
