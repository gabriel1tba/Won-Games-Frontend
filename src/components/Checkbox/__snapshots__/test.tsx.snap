// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Checkbox /> should render the heading 1`] = `
<main
  class=""
>
  <h1>
    Checkbox
  </h1>
</main>
`;
